### PR TITLE
CT-2588 Move open cases additional

### PR DESCRIPTION
**Pull request by Polsonby**
## Description
We found some edge cases through testing; this PR addresses situations where users might not get moved along with the team, making the tests and code work for business units with "code" set and without, and for trigger cases ensuring that the approving team stays the same after the move.  
 
## Self-review checklist

* [x] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
 
 
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2292
 
### Manual testing instructions
Move a team :) 

All the users should still be there, and cases belonging to them should still be viewable (was throwing an exception) - check trigger and non-trigger cases.

### CODE REVIEW

File: app/services/team_move_service.rb [hunk 1]
```diff
@@ -81,10 +82,11 @@ def link_old_team_to_new_team
     @team.moved_to_unit = @new_team
     @team.save
   end
-  
+
   def restore_new_team_name_to_original_name
     # New team gets original team name to retain consistency for the users
     @new_team.name = @team.original_team_name
+    @new_team.code = @new_team.code.sub(/-NEW$/, "") unless @team.code.blank?
```
Comment (mseedat-moj) [inline app/services/team_move_service.rb]: Does this need to be gsub or case insensitive?

**Review by mseedat-moj — APPROVED** [review]
Great work, minor question

File: app/services/team_move_service.rb (hunk 1 above)
Comment (Polsonby) [reply app/services/team_move_service.rb]: Great question! I suck at regex, but I don't think so... the idea is to only replace the pattern if found at the end of the string so I think it's ok as sub rather than gsub. Similarly, we're adding the -NEW in uppercase so I only want to replace it if it's uppercase
